### DISCUSSION TITLE
Error: Duplicate request initiated.

### DISCUSSION THREAD

**Discussion by vladparl** [pactumjs/pactum#96] [Q&A] [answered]
Hello guys,

I am using pactum-cucumber-boilerplate project and trying to create scenario like this:

Scenario: Store email value in variable
    When make a GET request to /api/users/1
    Then response is received
    And response status is 200
    And response should have a json like
      """
      {
        "data":
            {
              "id":1,
              "email":"george.bluth@reqres.in",
              "first_name":"George",
              "last_name":"Bluth",
              "avatar":"https://reqres.in/img/faces/1-image.jpg"
            },
        "support":
            {
              "url":"https://reqres.in/#support-heading",
              "text":"To keep ReqRes free, contributions towards server costs are appreciated!"
              }
      }
      """
    And store response data.email in email
    When make a POST request with body to /api/login
    """
    {
      "email": "email",
      "password": "cityslicka"
    }
    """
    Then response is received
    And response status is 201

I am getting:  Error: Duplicate request initiated. Existing request - GET https://reqres.in/api/users/1

I have 2 questions: 
1. How to send 2 or more requests in one scenario?
2. How to get stored value from variable data.email and use in second scenario as email parameter?

**Comment by ASaiAnudeep** [comment, marked as answer]
> How to send 2 or more requests in one scenario?

To send more requests in a single scenario, just re-initiate the spec object in the appropriate step definition.

```js
When(/^I make a (.*) request with body to (.*)$/, function (method, endpoint, body) {
  spec = pactum.spec();
  spec.withMethod(method);
  spec.withPath(endpoint);
  spec.withBody(body);
});
```

More details [here](https://github.com/pactumjs/pactum/discussions/76)

> How to get stored value from variable data.email and use in second scenario as email parameter?

Use [data stores](https://pactumjs.github.io/#/integration-testing?id=stores).

```gherkin
Scenario: Get A User
    Given I make a GET request to /api/users/1
     When I receive a response
     Then I expect response should have a status 200
      And I store response at data.email as UserEmail

Scenario: List Users
    Given I make a GET request to /api/users
      And I set query param email to $S{UserEmail}
     When I receive a response
     Then I expect response should have a status 200
```

From the above example the email id from the server is saved in pactum's data-store. You can refer them in the next scenario as `$S{UserEmail}`

**Reply by vladparl** [reply]
Thank you so much ASaiAnudeep. Your answer was very helpful and is really appreciated :)

**Comment by vladparl** [comment]
Currently I have a function:

When(/^make a (.*) request to (.*)$/, function (method, endpoint) {
  spec = new spec();
  spec [method.toLowerCase()] (endpoint);
  spec.withMethod(method);
  spec.withPath(endpoint);
});

But get error message:

 × When make a GET request to /api/breeds/image/random # features\support\steps.js:39
       TypeError: spec is not a constructor
           at World.<anonymous> (C:\Users\vlade\Desktop\Projects\BDD_Rest_Api_Testing\features\support\steps.js:40:10)

What I am doing wrong?

**Reply by leelaprasadv** [reply]
Do a`pactum.spec();` instead of `new spec()`

Also declare the spec object variable on top of your steps.js

and in the `When`create a spec 
`spec = pactum.spec();`

More reference: https://pactumjs.github.io/#/api-testing?id=testing-style
Section tab: Breaking (cucumber)

**Reply by ASaiAnudeep** [reply]
My bad. Updated the original answer.
Thanks @leelaprasadv for the prompt response.
